Explictly define the default assignment operator for path

I am getting the following warning when compiling with Clang 13. 
```
In file included from path/to/rcpputils/src/filesystem_helper.cpp:39:
path/to/rcpputils/include/rcpputils/filesystem_helper.hpp:94:20: warning: definition of implicit copy assignment operator for 'path' is deprecated because it has a user-declared copy constructor [-Wdeprecated-copy]
  RCPPUTILS_PUBLIC path(const path & p) = default;
                   ^
path/to/rcpputils/src/filesystem_helper.cpp:197:14: note: in implicit copy assignment operator for 'rcpputils::fs::path' first required here
      parent = *it;
             ^
1 warning generated.
---
```
The warning and solution seem pretty straightforward and it looks like the warning can be resolved by explicitly defining the operator in `filesystem_helper.hpp` as follows:
```cpp
RCPPUTILS_PUBLIC path& operator=(const path & ) = default;
```

--- a/include/rcpputils/filesystem_helper.hpp
+++ b/include/rcpputils/filesystem_helper.hpp
@@ -94,6 +94,13 @@ public:
   RCPPUTILS_PUBLIC path(const path & p) = default;
 
   /**
+   * \brief Copy assignment operator.
+   *
+   * \return Reference to the copied path.
+   */
+  RCPPUTILS_PUBLIC path & operator=(const path &) = default;
+
+  /**
    * \brief Get the path delimited using this system's path separator.
    *
    * \return The path as a string

